ui/raidboss/data/06-ew/ultimate/dragonsongs_reprise_ultimate.ts

```diff
@@ -829,14 +847,301 @@ const triggerSet: TriggerSet<Data> = {
+      // Ignore targetIsYou() incase player misses stack
+      condition: (data) => data.eyeOfTheTyrantCounter === 2,
+      suppressSeconds: 1,
+      infoText: (data, _matches, output) => {
+        const num = data.diveFromGraceNum[data.me];
+        if (!num) {
+          console.error(`DFG Tower 3 Reminder: missing number: ${JSON.stringify(data.diveFromGraceNum)}`);
+          return;
+        }
+        // Call Tower 3 Soak for num1 (based on previous position)
+        if (num === 1 && data.diveFromGracePreviousPosition[data.me] === 'middle')
+          return output.southTower!();
+        // Call Tower 3 Soak for num2s (based on previous position)
+        if (num === 2) {
+          if (data.diveFromGracePreviousPosition[data.me] === 'west')
+            return output.westTower!();
+          if (data.diveFromGracePreviousPosition[data.me] === 'east')
+            return output.eastTower!();
+        }
+        // If failed to get positions, call Towers in general
+        if (num !== 3)
+          return output.circleTowers!();
+      },
+      outputStrings: {
+        southTower: {
+          en: 'South Tower',
+        },
+        circleTowers: {
+          en: 'Towers (all circles)',
+        },
+        westTower: {
+          en: 'West Tower',
+        },
+        eastTower: {
+          en: 'East Tower',
+        },
+      },
+    },
     {
       id: 'DSR Darkdragon Dive Single Tower',
       type: 'Ability',
       netRegex: NetRegexes.ability({ id: '6711', source: 'Nidhogg' }),
       condition: Conditions.targetIsYou(),
       suppressSeconds: 1,
-      infoText: (_data, _matches, output) => output.text!(),
+      infoText: (data, _matches, output) => {
+        const num = data.diveFromGraceNum[data.me];
+        if (!num) {
+          console.error(`DFG Dive Single Tower: missing number: ${JSON.stringify(data.diveFromGraceNum)}`);
+          return output.text!();
+        }
+        if (data.eyeOfTheTyrantCounter === 1) {
+          // Number 1s Stack after bait
+          if (num === 1)
+            return output.baitThenStack!();
+          // Number 3s Dive position reminder after bait (in case different from Tower)
+          if (num === 3) {
+            if (data.diveFromGraceHasArrow[3]) {
+              if (data.diveFromGraceDir[data.me] === 'circle')
+                return output.baitThenSouthDive!();
+              if (data.diveFromGraceDir[data.me] === 'up')
+                return output.baitThenUpArrowDive!();
+              if (data.diveFromGraceDir[data.me] === 'down')
+                return output.baitThenDownArrowDive!();
+              // Unknown debuff
+              return output.text!();
+            }
+            return output.baitThenCirclesDive!();
+          }
+        }
+        return output.text!();
+      },
+      run: (data) => data.waitingForGeirskogul = true,
       outputStrings: {
+        baitThenStack: {
+          en: 'Bait => Stack',
```

Comment:
I got a "Bait => Stack" as the call for my 3 tower bait as a south 1 circle with arrow.  I'll see if I can repro in the emulator and see what's wrong.

Reply:
That seems odd. Maybe a order of operations problem with data.eyeOfTheTyrantCounter = data.eyeOfTheTyrantCounter ?? 0 + 1 in the timeline trigger where it is always 1.

Reply:
typescript sim shows that to be the case, so I pushed parenthesis that resolves the trigger not incrementing properly.